### PR TITLE
runtime: replace use of `std::mutex` with `StaticMutex`

### PR DESCRIPTION
**Pull request by compnerd**
Replace the `static std::mutex` with Swift's `StaticMutex` so that we
avoid triggering an error due to an exit time destructor.  Repairs the
build after #38562.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves SR-NNNN.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
CC: @mikeash @al45tair

**Review by mikeash — APPROVED** [review]

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Comment by compnerd** [comment]
@swift-ci please smoke test macOS platform

**Comment by compnerd** [comment]
@swift-ci please smoke tes Lnux platform

**Comment by compnerd** [comment]
@swift-ci please smoke test Linux platform

**Comment by compnerd** [comment]
@swift-ci please clean test Linux platform

File: stdlib/public/runtime/Errors.cpp [hunk 1]
```diff
@@ -91,21 +91,19 @@ static bool getSymbolNameAddr(llvm::StringRef libraryName,
   // demangle with swift. We are taking advantage of __cxa_demangle actually
   // providing failure status instead of just returning the original string like
   // swift demangle.
+  static StaticMutex mutex;
```
Comment (al45tair) [inline stdlib/public/runtime/Errors.cpp:94]: The `#if defined(_WIN32)` needs to come before this line, I think.

Comment (compnerd) [reply stdlib/public/runtime/Errors.cpp:94]: Ah, right you are - it seems that I didn't move the macros around properly

**Comment by compnerd** [comment]
@swift-ci please test

**Pull request merged**